Source/WebCore/loader/FrameLoader.cpp

```diff
@@ -1219,6 +1219,32 @@ static NavigationNavigationType determineNavigationType(FrameLoadType loadType,
     return NavigationNavigationType::Push;
 }
 
+// https://html.spec.whatwg.org/multipage/browsing-the-web.html#url-and-history-update-steps
+void FrameLoader::updateURLAndHistory(const URL& newURL, RefPtr<SerializedScriptValue>&& stateObject, NavigationHistoryBehavior historyHandling)
+{
+    ASSERT(m_frame->document());
+
+    if (documentLoader()->isInitialAboutBlank())
+        historyHandling = NavigationHistoryBehavior::Replace;
+
+    CheckedRef history = m_frame->checkedHistory();
+
+    // FIXME: Validate that stateObject deserializes without exception.
```

Comment:
I don’t understand the meaning of the word “validate” here. Can we make it clearer what needs to be done?

Reply:
I rephrased it, basically it is one of the algorithm steps.